docket/api.py

```diff
@@ -1,20 +1,28 @@
 from flask import Flask, Blueprint, Response
-from flask_restful import Resource, Api
+from flask_restful import Api
 
-from resources.pcap import PcapApi, PcapUri, RawQuery
-from resources.stats import Stats
+from resources.query import QueryRequest, ApiRequest, RawRequest
+
+from config import Config
 
 # Declare the blueprint
 api_bp = Blueprint('api', __name__)
-api = Api(api_bp)
+api    = Api(api_bp)
 
 # Add resources
-api.add_resource(PcapApi, '/api/')
-api.add_resource(PcapUri, '/pcap/<path:path>')
-api.add_resource(RawQuery, '/query' )
-api.add_resource(Stats, '/stats/', '/stats/<string:sensors>', endpoint='stats')
+WEB_ROOT = Config.get('WEB_ROOT', '/')
+# consider 'login_required' for queries
+api.add_resource(RawRequest, WEB_ROOT+'raw/<string:query>' )     # ex: /raw/host+1.2.3.4+port+80
```

Comment:
Ohhhh....it took me a while to figure out what this `WEB_ROOT` was doing, and it made sense. However, with Flask, you can actually handle this in a single place with [blueprints](http://flask.pocoo.org/docs/0.12/blueprints/). This can be done on a single line [here](https://github.com/rocknsm/docket/blob/master/docket/api.py#L19).

Please remove the rest of the string concatenations.

Reply:
I reworked this as you suggested.